stepfunctions/templates/consignment_export_definition.json.tpl

```diff
@@ -31,7 +39,38 @@
           ]
         }
       },
+      "Next": "Task complete notification"
+    },
+    "Task complete notification": {
+      "Type": "Task",
+      "Resource": "arn:aws:states:::sns:publish",
+      "Parameters": {
+        "Message": {
+          "consignmentId.$": "$.Overrides.ContainerOverrides[0].Environment[0].Value",
+          "success": true
+        },
+        "TopicArn": "${sns_topic}"
+      },
+      "End": true
+    },
+    "Handle escaped JSON from error cause": {
+      "Type": "Pass",
+      "Parameters": {
+        "Cause.$": "States.StringToJson($.Cause)"
+      },
+      "Next": "Task failed notification"
+    },
+    "Task failed notification": {
+      "Type": "Task",
+      "Resource": "arn:aws:states:::sns:publish",
+      "Parameters": {
+        "Message": {
+          "consignmentId.$": "$.Cause.Overrides.ContainerOverrides[0].Environment[0].Value",
```

Comment:
Do we need to pass in the rest of the Cause, so that it can be logged? Are there other relevant environment variables? Is there a message that would be helpful to know?

Reply:
Not that I can see. This is the full json
```json
{
  "Cause": {
    "Attachments": [
      {
        "Details": [
          {
            "Name": "subnetId",
            "Value": "subnet_id"
          },
          {
            "Name": "networkInterfaceId",
            "Value": "eni"
          },
          {
            "Name": "macAddress",
            "Value": ""
          },
          {
            "Name": "privateDnsName",
            "Value": ""
          },
          {
            "Name": "privateIPv4Address",
            "Value": ""
          }
        ],
        "Id": "43a1791c-833c-4722-8cb9-dc14d792acbb",
        "Status": "DELETED",
        "Type": "eni"
      }
    ],
    "Attributes": [],
    "AvailabilityZone": "regionb",
    "ClusterArn": "arn:aws:ecs:region:account_id:cluster/consignment_export_intg",
    "Connectivity": "CONNECTED",
    "ConnectivityAt": 1611269488246,
    "Containers": [
      {
        "ContainerArn": "arn:aws:ecs:region:account_id:container/1490de17-7f9d-4b83-b0c5-9ea2cfd307d7",
        "Cpu": "0",
        "ExitCode": 1,
        "GpuIds": [],
        "Image": "account.dkr.ecr.region.amazonaws.com/consignment-export:intg",
        "ImageDigest": "sha256:c7308c4c8efb5e93ca5a32320ee75312deecee23d78f67475bf2e3bf91d0149a",
        "LastStatus": "STOPPED",
        "Name": "consignmentexport",
        "NetworkBindings": [],
        "NetworkInterfaces": [
          {
            "AttachmentId": "43a1791c-833c-4722-8cb9-dc14d792acbb",
            "PrivateIpv4Address": ""
          }
        ],
        "RuntimeId": "87cd925ff47947af8124e18b45086219-2829816796",
        "TaskArn": "arn:aws:ecs:region:account_id:task/consignment_export_intg/87cd925ff47947af8124e18b45086219"
      }
    ],
    "Cpu": "512",
    "CreatedAt": 1611269484813,
    "DesiredStatus": "STOPPED",
    "ExecutionStoppedAt": 1611269515000,
    "Group": "family:consignment-export-intg",
    "InferenceAccelerators": [],
    "LastStatus": "STOPPED",
    "LaunchType": "FARGATE",
    "Memory": "1024",
    "Overrides": {
      "ContainerOverrides": [
        {
          "Command": [],
          "Environment": [
            {
              "Name": "CONSIGNMENT_ID",
              "Value": "9d26172d-1de1-484a-a5bc-d1b12ce7f158"
            }
          ],
          "EnvironmentFiles": [],
          "Name": "consignmentexport",
          "ResourceRequirements": []
        }
      ],
      "InferenceAcceleratorOverrides": []
    },
    "PlatformVersion": "1.4.0",
    "PullStartedAt": 1611269497021,
    "PullStoppedAt": 1611269510021,
    "StartedAt": 1611269513021,
    "StartedBy": "AWS Step Functions",
    "StopCode": "EssentialContainerExited",
    "StoppedAt": 1611269538689,
    "StoppedReason": "Essential container in task exited",
    "StoppingAt": 1611269525510,
    "Tags": [],
    "TaskArn": "arn:aws:ecs:region:account_id:task/consignment_export_intg/87cd925ff47947af8124e18b45086219",
    "TaskDefinitionArn": "arn:aws:ecs:region:account_id:task-definition/consignment-export-intg:4",
    "Version": 5
  }
}
```
The only thing that's relevant is the exit code but that's either 0 or 1 so there's not much we can get from it. All of the information about the failure is in the cloudwatch logs.

Reply:
Hmm, yeah, if there's nothing more specific than the usual "Essential container in task exited" then I guess we'd have to debug this by looking at the ECS logs anyway.